Source/Xamarin/Prism.DryIoc.Forms/PrismApplication.cs

```diff
@@ -0,0 +1,104 @@
+        {
+            Logger = CreateLogger();
+
+            ModuleCatalog = CreateModuleCatalog();
+            ConfigureModuleCatalog();
+
+            Container = CreateContainer();
+            NavigationService = CreateNavigationService();
+            ConfigureContainer();
+
+            RegisterTypes();
+
+            InitializeModules();
+
+            OnInitialized();
+        }
+
+        /// <summary>
+        /// Create a default instance of <see cref="IContainer"/> with <see cref="Rules"/> created in <see cref="CreateContainerRules"/>
+        /// </summary>
+        /// <returns>An instance of <see cref="IContainer"/></returns>
+        protected virtual IContainer CreateContainer()
+        {
+            var rules = CreateContainerRules();
+            return new Container(rules);
+        }
+
+        /// <summary>
+        /// Create <see cref="Rules"/> to alter behavior of <see cref="IContainer"/>
+        /// </summary>
+        /// <remarks>
+        /// Default rule is to consult <see cref="Xamarin.Forms.DependencyService"/> if the requested type cannot be inferred from <see cref="Container"/>
+        /// </remarks>
+        /// <returns>An instance of <see cref="Rules"/></returns>
+        protected virtual Rules CreateContainerRules()
+        {
+            return UnknownServiceResolverRule.DependencyServiceResolverRule;
+        }
+
+        protected virtual void ConfigureContainer()
+        {
+            Container.RegisterInstance(Logger);
+            Container.RegisterInstance(ModuleCatalog);
+            Container.RegisterInstance(NavigationService);
+            Container.Register<IModuleManager, ModuleManager>(Reuse.Singleton);
+            Container.Register<IModuleInitializer, DryIocModuleInitializer>(Reuse.Singleton);
+            Container.Register<IEventAggregator, EventAggregator>(Reuse.Singleton);
+            Container.Register<IDependencyService, DependencyService>(Reuse.Singleton);
+            Container.Register<IPageDialogService, PageDialogService>(Reuse.Singleton);
+        }
+
+        protected override void InitializeModules()
+        {
+            if (ModuleCatalog.Modules.Any())
+            {
+                var manager = Container.Resolve<IModuleManager>();
+                manager.Run();
+            }
+        }
+
+        protected override INavigationService CreateNavigationService()
+        {
+            return new DryIocPageNavigationService(new ApplicationProvider(), Container);
+        }
+
+        protected override void ConfigureViewModelLocator()
+        {
+            ViewModelLocationProvider.SetDefaultViewModelFactory((view, type) =>
+            {
+                var page = view as Page;
+                if (page != null)
+                {
+                    var navigationService = Container.Resolve<INavigationService>();
+                    ((IPageAware)navigationService).Page = page;
+                }
+                return Container.Resolve(type);
```

Comment:
The navigation service cannot be a singleton.  We have to create a new instance each time and then override the parameters of the VM ctor to inject the correct instance.  Can this be done in DryIoc?